clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -298,7 +299,7 @@ static KafkaAdminClient createInternal(AdminClientConfig config, TimeoutProcesso
                     config.getLong(AdminClientConfig.METADATA_MAX_AGE_CONFIG), true);
             List<MetricsReporter> reporters = config.getConfiguredInstances(AdminClientConfig.METRIC_REPORTER_CLASSES_CONFIG,
                 MetricsReporter.class);
-            Map<String, String> metricTags = Collections.singletonMap("client-id", clientId);
+            Map<String, String> metricTags = Collections.singletonMap("client-id", Sanitizer.sanitize(clientId));
```

Comment:
Since there are two calls to `sanitize` here, perhaps worth doing once and reusing?

Reply:
Here to avoid the 2nd call, we'd have to pass sanitizedClientId as an argument to the private constructor. Is it what you prefer ?

Reply:
Personally, yes, I prefer one call, but I leave it up to you.